dev/com.ibm.ws.org.jboss.weld/src/org/jboss/weld/resources/FileLogger.java

```diff
@@ -0,0 +1,707 @@
+            String fileName = getProperty(FILE_PROPERTY_NAME, "JHotspot");
+            String fileExt = getProperty(FILE_EXT_PROPERTY_NAME, ".log");
+            String prefix = getProperty(PREFIX_PROPERTY_NAME, "JBH: ");
+            boolean autoflush = getProperty(AUTOFLUSH_PROPERTY_NAME, false);
+
+            return new FileLogger(dirName, fileName, fileExt, prefix, autoflush);
+        }
+
+        //
+
+        public static final String WLP_INSTALL_PROPERTY_NAME = "wlp.install.dir";
+        public static final String WLP_SERVER_PROPERTY_NAME = "wlp.server.name";
+        public static final String WLP_LOG1_PROPERTY_NAME = "com.ibm.ws.logging.log.directory";
+        public static final String WLP_LOG2_PROPERTY_NAME = "LOG_DIR";
+
+        public static final String DIR_PROPERTY_NAME         = "JBOSS_HOTSPOT_TRACE_DIR";
+        public static final String FILE_PROPERTY_NAME        = "JBOSS_HOTSPOT_TRACE_FILE";
+        public static final String FILE_EXT_PROPERTY_NAME    = "JBOSS_HOTSPOT_TRACE_EXT"; 
+        public static final String PREFIX_PROPERTY_NAME      = "JBOSS_HOTSPOT_TRACE_PREFIX";
+        public static final String AUTOFLUSH_PROPERTY_NAME   = "JBOSS_HOTSPOT_TRACE_AUTOFLUSH";
+        public static final String PATTERN_PROPERTY_NAME     = "JBOSS_HOTSPOT_TRACE_PATTERN";
+
+        public static final String[] PROPERTY_NAMES = {
+            DIR_PROPERTY_NAME,
+            FILE_PROPERTY_NAME,
+            FILE_EXT_PROPERTY_NAME,
+            PREFIX_PROPERTY_NAME,
+            AUTOFLUSH_PROPERTY_NAME,
+            PATTERN_PROPERTY_NAME
+        };
+
+        public FileLoggerProperties() {
+            this(null);
+        }
+
+        public FileLoggerProperties(Properties properties) {
+            this.overrideProperties = properties;
+            this.overridePattern = ( (properties == null) ? null : properties.getProperty(PATTERN_PROPERTY_NAME) );
+
+            this.wlpHome = System.getProperty(WLP_INSTALL_PROPERTY_NAME);
+            this.wlpName = System.getProperty(WLP_SERVER_PROPERTY_NAME);
+            this.log1Home = System.getProperty(WLP_LOG1_PROPERTY_NAME);
+            this.log2Home = System.getProperty(WLP_LOG2_PROPERTY_NAME);
+
+            this.logHome = selectLogHome();
+
+            System.out.println("JBH: Install Home [ " + WLP_INSTALL_PROPERTY_NAME + " ] [ " + wlpHome + " ]");
+            System.out.println("JBH: Server Name [ " + WLP_SERVER_PROPERTY_NAME + " ] [ " + wlpName + " ]");
+            System.out.println("JBH: Log1 Home [ " + WLP_LOG1_PROPERTY_NAME + " ] [ " + log1Home + " ]");
+            System.out.println("JBH: Log2 Home [ " + WLP_LOG2_PROPERTY_NAME + " ] [ " + log2Home + " ]");
+            System.out.println("JBH: Log Home [ " + logHome + " ]");
+
+            Properties useProperties = new Properties();
+
+            for ( String propertyName : PROPERTY_NAMES ) {
+                String propertyValue = System.getProperty(propertyName);
```

Comment:
This also should be using doPriv to get System properties

Reply:
Added doPriv.